src/haddock/gear/clean_steps.py

```diff
@@ -0,0 +1,87 @@
+"""Clean module output."""
+import gzip
+import shutil
+import tarfile
+from multiprocessing import Pool
+from pathlib import Path
+
+from haddock import log
+from haddock.libs.libio import (
+    archive_files_ext,
+    compress_files_ext,
+    glob_folder,
+    remove_files_with_ext,
+    )
+
+
+UNPACK_FOLDERS = []
+
+
+def clean_output(path, ncores=1):
+    """Perform operations after the run."""
+    log.info(f"Cleaning output for {str(path)!r} using {ncores} cores.")
+    # add any formats generated to
+    # `unpack_compressed_and_archived_files` so that the
+    # uncompressing routines when restarting the run work.
+    files_to_archive = ['seed', 'inp', 'out', 'con']
+    for fta in files_to_archive:
+        found = archive_files_ext(path, fta)
+        if found:
+            remove_files_with_ext(path, fta)
+
+    files_to_compress = ['pdb', 'psf']
+    for ftc in files_to_compress:
+        found = compress_files_ext(path, ftc, ncores=ncores)
+        if found:
+            remove_files_with_ext(path, ftc)
+
+
+def unpack_compressed_and_archived_files(folders, ncores):
+    """Unpack compressed and archived files in a folders."""
+    global UNPACK_FOLDERS
+    UNPACK_FOLDERS.clear()
+
+    for folder in folders:
+        gz_files = glob_folder(folder, '.gz')
+        tar_files = glob_folder(folder, '.tar')
```

Comment:
Shouldn't you rather have `.tgz` as extension if you generate compressed tar archives (giving the `z` option to tar)?